Makefile

```diff
@@ -1251,14 +1251,17 @@ runtime/ld.conf: $(ROOTDIR)/Makefile.config
 	$(V_GEN)echo "$(STUBLIBDIR)" > $@ && \
 	echo "$(LIBDIR)" >> $@
 
+PRIMITIVES_NEW := runtime/primitives$(shell echo "$$PPID").new
+
 # To speed up builds, we avoid changing "primitives" when files
 # containing primitives change but the primitives table does not
 runtime/primitives: runtime/gen_primitives.sh \
   $(shell runtime/gen_primitives.sh $(runtime_BYTECODE_C_SOURCES) \
-                    > runtime/primitives.new; \
-                    cmp -s runtime/primitives runtime/primitives.new || \
-                    echo runtime/primitives.new)
-	$(V_GEN)cp runtime/primitives.new $@
+                    > $(PRIMITIVES_NEW); \
+                    { cmp -s runtime/primitives $(PRIMITIVES_NEW) && \
+                        rm -f $(PRIMITIVES_NEW) ; } || \
+                    echo $(runtime_BYTECODE_C_SOURCES))
```

Comment:
Why did you change what was `echoed` here?

Reply:
That's part of trying to avoid filling `runtime/` with lots of `primitives23434525.new` files - so the recipe now either deletes the file (if it was identical) or does an `mv` rather than a `cp`. I could have just put `echo $(PRIMITIVES_NEW)`, but I figured it was confusing (and possibly buggy) to be referring to a dependency which would then be deleted. If the primitives list has been changed, at least one of those files must be newer, so I switched to echoing them instead (it's a nice consequence of the recent change to pass that list of files to the script, rather than hard-coding it)